core/libcamera_encoder.hpp

```diff
@@ -72,6 +72,8 @@ class LibcameraEncoder : public LibcameraApp
 			std::lock_guard<std::mutex> lock(encode_buffer_queue_mutex_);
 			if (encode_buffer_queue_.empty())
 				throw std::runtime_error("no buffer available to return");
+			auto completed_request = encode_buffer_queue_.front();
+			encoder_->metadataReady(completed_request->metadata);
```

Comment:
I'm not sure about making it the Encoder's job to output metadata, it doesn't quite feel like a thing an Encoder should do. Can we make this a method of the output object?